src/equipment/equipment.service.ts

```diff
@@ -0,0 +1,126 @@
+import { HttpStatus, Injectable } from '@nestjs/common';
+import { Equipment } from '../schemas/requestable/equipment';
+import handlePromise from '../utils/promise';
+import { BackendException } from '../shared/backend.exception';
+import { Model, Types } from 'mongoose';
+import { EquipmentController } from './equipment.controller';
+import { EquipmentModule } from './equipment.module';
+import { InjectModel } from '@nestjs/mongoose';
+import { EquipmentdbService } from './equipment-db.service'
+
+
+@Injectable()
+export class EquipmentService {
+  constructor( 
+    @InjectModel(Equipment.name)
+    private EquipmentModel: Model<Equipment>,
+    private readonly dbEquipment: EquipmentdbService,
+
+  ) { }
+
+  async createEquipment(equipment: Equipment) {
+    const [newequipment , err ] = await   handlePromise(this.dbEquipment.createEquipment(equipment))
```

Comment:
en este caso newEquipment no es necesario podemos obviarlo directamente
` const [, err ] = await   handlePromise(this.dbEquipment.createEquipment(equipment))`